src/content-script.ts

```diff
@@ -4,14 +4,47 @@ import Mustache from "mustache";
 import { createElement } from "./elements";
 import { isBodyWithReplies, isRepositoryDetails } from "./validations";
 
-(async function main() {
+const regexPathName = /\/.+\/.+\/(issues|pull)\/\d+/;
+const replySummaryId = "saved-reply-new_comment_field";
+const footprint = "refined-saved-replies";
+
+// hack for detecting page change
+function addObserver(callback: MutationCallback) {
+	const options = { childList: true, subtree: true };
+
+	const observer = new MutationObserver(callback);
+
+	observer.observe(document.body, options);
```

Comment:
> `document.body`

[Performance] So this is going to keep firing whenever anything in the DOM changes... that's a lot of firing! A few suggestions:

* Very unfortunate that I/we couldn't find a native way to listen to the URL on stream. Maybe I'm missing something? Surely there's got to be a way...
    * Honestly, if not, it'd be super reasonable to wrap whatever logic this PR ends up using into a standalone package 😄 
* Maybe listening to the `<turbo-frame>` element itself would be fewer events? That one _seems_ to have a `src` attribute that changes whenever the page URL does.
* Your original strategy of using a background script seems to work? I'm not against background scripts overall, just asking users for more permissions. So maybe this would be a good fallback if we can't get DOM mutation observing to work?

Reply:
Exciting news @JoshuaKGoldberg!

After investigating Github UI, I discovered github dispatches a series of events for the turbo "soft nav" navigation. 

The extension works perfectly when listening to event 'soft-nav:success' but I do want to continue investigating and figure out what which of the events below is more appropriate. 

Should I remove the previous 2 commits for a cleaner PR to merge?

```
'soft-nav:initial',
'soft-nav:start',
'soft-nav:success',
'soft-nav:error',
'soft-nav:frame-update',
'soft-nav:end',
'soft-nav:render',
'soft-nav:progress-bar:start',
'soft-nav:progress-bar:end'
'soft-nav:external:initial',
'soft-nav:external:start',
'soft-nav:external:success',
'soft-nav:external:error',
'soft-nav:external:render'
```

Reply:
Aha awesome! I much prefer this 😄 great find!